design/019-local_development_flow.md

```diff
@@ -0,0 +1,62 @@
+
+The local ``conan build`` command was intended to easily call the current recipe ``build()`` method, directly in the user folder. The main problem is that the ``build()`` method needs some state: the settings, the options. So ``conan build`` started to use a local ``conaninfo.txt`` file that was generated in a previous ``conan install`` command. Also, it was necessary to have the ``deps_cpp_info`` information, so that was serialized to a ``conanbuildinfo.txt`` also in a previous ``conan install`` command. Finally, the name, version, user, channel of the recipe was also necessary, and that is recovered from a serialized ``graph_info.json`` file, from the previous ``conan install``. All the state that might be necessary for a build needs to be recovered from a serialized representation from the previous ``conan install``. This makes this approach fragile, as any missing information or small bug in serialization/deserialization causes errors and problems in that local build. It is also frequently confusing for users, from a UX perspective, that they cannot ``conan build -s build_type=Debug``, because build can only use that information from a previous saved state.
+
+This ``conan build`` command was very necessary because the integration with build systems relied a lot on the helpers called in the recipe ``build()`` method. With the new ``generate()`` method and the toolchain, the goal is to achieve a very transparent integration with the build systems, in which the user can more easily call the build system and achieve exactly the same build as will happen in the Conan cache if the package is created with ``conan create``. Even in that case, the ``conan build`` command is useful for abstracting away the details to call the build system, which is relevant in CI for consumer projects that could use different build systems, and also for developers, specially in that case of having multiple build systems, each one with its own way to be called in command line. The core of the proposal is that the ``conan build`` shouldn't be mandatory, and there should be a relatively friendly UX way to achieve the same result if calling the build system directly.
+
+
+## Proposal
+
+- To remove the ``conan package`` command and propose to use ``conan export-pkg`` to test/debug the ``package()`` functionality if necessary.
+- Keep working, providing in ``conan.tools.xxxx`` new tools and toolchains a full working set of helpers to use in the ``generate()`` method, that operate creating files for the user convenience, so users can build directly with their native tools, instead of relying on the ``conan build` command.
```

Comment:
Typo 🙊 

```suggestion
- Keep working, providing in ``conan.tools.xxxx`` new tools and toolchains a full working set of helpers to use in the ``generate()`` method, that operate creating files for the user convenience, so users can build directly with their native tools, instead of relying on the ``conan build`` command.
```